plugins/modules/aci_epg_subnet.py

```diff
@@ -0,0 +1,473 @@
+
+# Copyright: (c) 2023, Gaspard MICOL (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "certified"}
+
+DOCUMENTATION = r"""
+---
+module: aci_epg_subnet
+short_description: Manage EPG Subnets (fv:Subnet)
+description:
+- Manage EPG Subnets on Cisco ACI fabrics.
+options:
+  tenant:
+    description:
+    - The name of the Tenant.
+    type: str
+    aliases: [ tenant_name ]
+  ap:
+    description:
+    - Name of an existing application network profile, that will contain the EPGs.
+    type: str
+    aliases: [ app_profile, app_profile_name ]
+  epg:
+    description:
+    - Name of the end point group.
+    type: str
+    aliases: [ epg_name, name ]
+  description:
+    description:
+    - The description for the Subnet.
+    type: str
+    aliases: [ descr ]
+  enable_vip:
+    description:
+    - Determines if the Subnet should be treated as a VIP (virtual IP address).
+    - The APIC defaults to C(false) when unset during creation.
+    type: bool
+  gateway:
+    description:
+    - The IPv4 or IPv6 gateway address for the Subnet.
+    type: str
+    aliases: [ gateway_ip ]
+  mask:
+    description:
+    - The subnet mask for the Subnet.
+    - This is the number associated with CIDR notation.
+    - For IPv4 addresses, accepted values range between C(0) and C(32).
+    - For IPv6 addresses, accepted Values range between C(0) and C(128).
+    type: int
+    aliases: [ subnet_mask ]
+  nd_prefix_policy:
+    description:
+    - The IPv6 Neighbor Discovery Prefix Policy to associate with the Subnet.
+    type: str
+  preferred:
+    description:
+    - Determines if the Subnet is preferred over all available Subnets. Only one Subnet per Address Family (IPv4/IPv6) can be preferred in the Bridge Domain.
+    - The APIC defaults to C(false) when unset during creation.
+    type: bool
+  route_profile:
+    description:
+    - The Route Profile to the associate with the Subnet.
+    type: str
+  route_profile_l3out:
+    description:
+    - The L3Out that contains the associated Route Profile.
+    type: str
+  scope:
+    description:
+    - Determines the scope of the Subnet.
+    - The C(private) option only allows communication with hosts in the same VRF.
+    - The C(public) option allows the Subnet to be advertised outside of the ACI Fabric, and allows communication with hosts in other VRFs.
+    - The shared option limits communication to hosts in either the same VRF or the shared VRF.
+    - The value is a list of options, C(private) and C(public) are mutually exclusive, but both can be used with C(shared).
+    - The APIC defaults to C(private) when unset during creation.
+    type: list
+    elements: str
+    choices:
+      - private
+      - public
+      - shared
```

Comment:
To be consistent in the documentation.
```suggestion
    choices: [ private, public, shared ]
```